src/app/components/loaders/btc-balance-loader.tsx

```diff
@@ -1,33 +1,29 @@
-import type { BtcBalance } from '@leather.io/models';
+import type { AccountQuotedBtcBalance } from '@leather.io/services';
 import { BtcAvatarIcon } from '@leather.io/ui';
 
-import { useBtcCryptoAssetBalanceNativeSegwit } from '@app/query/bitcoin/balance/btc-balance-native-segwit.hooks';
+import { useNativeSegwitBtcAccountBalance } from '@app/query/bitcoin/balance/btc-balance.hooks';
 
 import { CryptoAssetItemError } from '../crypto-asset-item/crypto-asset-item-error';
 import { CryptoAssetItemPlaceholder } from '../crypto-asset-item/crypto-asset-item-placeholder';
 
-interface BtcBalanceLoaderProps {
-  address: string;
-  children(balance: BtcBalance): React.ReactNode;
-}
-export function BtcBalanceLoader({ address, children }: BtcBalanceLoaderProps) {
-  const { balance } = useBtcCryptoAssetBalanceNativeSegwit(address);
-  return children(balance);
-}
-
 interface BtcAssetItemBalanceLoaderProps {
-  address: string;
+  accountIndex: number;
   children(
-    balance: BtcBalance,
+    balance: AccountQuotedBtcBalance,
     isLoading: boolean,
     isLoadingAdditionalData: boolean
   ): React.ReactNode;
 }
-export function BtcAssetItemBalanceLoader({ address, children }: BtcAssetItemBalanceLoaderProps) {
-  const { balance, filteredUtxosQuery, isLoading, isLoadingAdditionalData } =
-    useBtcCryptoAssetBalanceNativeSegwit(address);
+export function BtcAssetItemBalanceLoader({
+  accountIndex,
+  children,
+}: BtcAssetItemBalanceLoaderProps) {
+  const nativeSegwitBalance = useNativeSegwitBtcAccountBalance(accountIndex);
+  const isLoading = nativeSegwitBalance.state === 'loading';
```

Comment:
I see now we handle loading here so maybe it's overkill to add it as suggested [here](https://github.com/leather-io/extension/pull/6337/files#r2297266231). It would be good if we can avoid excess `createMoney(0` though